MSAL/src/native_auth/public/MSALNativeAuthUserAccountResult+Internal.swift

```diff
@@ -70,6 +70,8 @@ extension MSALNativeAuthUserAccountResult {
                 let accessTokenResult = MSALNativeAuthTokenResult(accessToken: result.accessToken,
                                                                   scopes: result.scopes,
                                                                   expiresOn: result.expiresOn)
+                self.rawIdToken = result.idToken
```

Comment:
We already fixed this issue in the #2141 PR. In that PR why added some tests, I'm wondering why those tests are not failing. We need to find a better way to test this class. I suggest adding a unit test to check that the idToken and account are updated. To make this class testable I suggested a different design here:
https://github.com/AzureAD/microsoft-authentication-library-for-objc/pull/2159#discussion_r1610149398
I believe it's time to make this change.

Reply:
Done